### PR TITLE
Disable zookeeper when running single replica

### PR DESCRIPTION
**Pull request by johnplacek**
1. Disabled zookeeper pack when running a single replica (Zookeeper not required).
2. Updated imagePullSecret template to include registry to allow it to pull from a private repository 
3. Added fullname template to allow user to override kubernetes resource names with fullnameOverride

### CODE REVIEW

File: charts/stardog/templates/configmap.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ data:
   {{- (.Files.Glob "files/log4j2.xml").AsConfig | nindent 2 }}
 
   {{- (.Files.Glob "files/stardog.properties").AsConfig | nindent 2 }}
-    pack.enabled=true
+    pack.enabled={{ if (eq (.Values.replicaCount | int) 1) }} false {{ else }} true {{ end }}
```
Comment (pdmars) [inline charts/stardog/templates/configmap.yaml:17]: Could you add quick check to the smoke tests to verify this? E.g. right after:

https://github.com/stardog-union/helm-charts/blob/master/tests/smoke.sh#L176

Create a single node deployment with this value set and then check that only a single Stardog pod starts and works and that no ZKs start?

File: charts/stardog/templates/statefulset.yaml [hunk 2]
```diff
@@ -73,9 +73,9 @@ spec:
           echo "Using existing zookeeper"
         {{ end }}
       containers:
-      - name: {{ .Release.Name }}-stardog
+      - name: {{ include "stardog.fullname" . }}
         imagePullPolicy: {{ .Values.image.pullPolicy }}
-        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        image: {{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}
```
Comment (pdmars) [inline charts/stardog/templates/statefulset.yaml:78]: The tests fail to start stardog with:

```
  Warning  InspectFailed     14s (x2 over 29s)      kubelet, default-a2ebce75-00cd-4a57-9d5a-b94bc233b63f  Failed to apply default image tag "https://registry.hub.docker.com/v1/repositories/stardog/stardog:latest": couldn't parse image reference "https://registry.hub.docker.com/v1/repositories/stardog/stardog:latest": invalid reference format
  Warning  Failed            14s (x2 over 29s)      kubelet, default-a2ebce75-00cd-4a57-9d5a-b94bc233b63f  Error: InvalidImageName
```

I think the registry is only needed in the image secret string and when those credentials are used the `{{ .Values.image.repository }}:{{ .Values.image.tag }}` will pull from there.

File: charts/stardog/templates/configmap.yaml (hunk 1 above)
Comment (buzztroll) [reply charts/stardog/templates/configmap.yaml:17]: I would prefer that this be based on a flag and not `replicaCount == 1`.  It is valid to have a cluster of 1.  Further what would happen in this case if someone had a 3 node cluster but brought it down to 1 for a an interim maintenance period?

File: charts/stardog/templates/statefulset.yaml (hunk 2 above)
Comment (johnplacek) [reply charts/stardog/templates/statefulset.yaml:78]: Done. I've tested it manually pulling from dockerhub and the private jfrog repo and it works.
I wasn't able to add a test for the private repo as it requires the password in the config.

File: charts/stardog/templates/configmap.yaml (hunk 1 above)
Comment (johnplacek) [reply charts/stardog/templates/configmap.yaml:17]: Done - created a cluster.enabled flag and a test to verify the behaviour.

File: tests/smoke.sh [hunk 3]
```diff
@@ -69,6 +71,33 @@ function helm_install_stardog_cluster() {
 	echo "Stardog Cluster installed."
 }
 
+function helm_install_single_node_stardog() {
+	echo "Installing Stardog Cluster"
```
Comment (pdmars) [inline tests/smoke.sh:75]: s/Stardog Cluster/single node Stardog

Comment (johnplacek) [reply tests/smoke.sh]: good catch! Done :-)

File: tests/smoke.sh [hunk 4]
```diff
@@ -69,6 +71,33 @@ function helm_install_stardog_cluster() {
...
+	             --set "replicaCount=1" \
+	             --set "zookeeper.enabled=false"
+	rc=$?
+
+	if [ ${rc} -ne 0 ]; then
+		echo "Helm install for Stardog Cluster failed, exiting"
+		exit ${rc}
+	fi
+
+	echo "Stardog Cluster installed."
```
Comment (pdmars) [inline tests/smoke.sh:98]: oops here too

**Comment by johnplacek** [comment]
Hi @pdmars could you generate a 1.0.1 release with the changes, please?

**Pull request merged**